pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -1793,3 +1929,128 @@ PJ_DEF(pj_status_t) pjmedia_sdp_neg_fmt_match(pj_pool_t *pool,
                             offer, o_fmt_idx, answer, a_fmt_idx, option);
 }
 
+static unsigned find_valid_pt(pj_pool_t *pool, media_pt_map *used_pt, 
+                              const pj_str_t *codec)
+{
+    unsigned pt_check;
+    unsigned i = 0;
+    unsigned used_pt_num = used_pt->med_pt_map_cnt;
+    unsigned max_pt_cnt = PJ_ARRAY_SIZE(used_pt->med_pt_map);
+    const pj_str_t telephone_event = pj_str("telephone-event");
+
+    if (pj_strncmp(codec, &telephone_event, telephone_event.slen) == 0)
+        pt_check = PJMEDIA_RTP_PT_TELEPHONE_EVENTS;
+    else
+        pt_check = START_DYNAMIC_PT;
+    
+    for (; i < used_pt_num && i < max_pt_cnt; ++i) {
+        if (pj_strcmp(codec, &used_pt->med_pt_map[i].codec) == 0) {
+            return used_pt->med_pt_map[i].pt;
+        }
+    }
+    i = 0;
+
+    while ((i < used_pt_num) && (i < max_pt_cnt) && (pt_check <= 127))
+    {
+        if (pt_check == used_pt->med_pt_map[i].pt) {
+            pt_check++;
+            i = 0;
+        } else {
+            i++;
+        }
+    }
+    if (pt_check > 127 || i == max_pt_cnt) {
+        /* No more available PT */
+        return 0;
+    }
+
+    used_pt->med_pt_map[used_pt->med_pt_map_cnt].pt = pt_check;
+    pj_strdup(pool, &used_pt->med_pt_map[used_pt->med_pt_map_cnt++].codec, codec);
```

Comment:
minor: this function is not just finding PT (=IMHO the map is read-only), it may assign a PT & update the map too, perhaps better rename it to something like `assign`?